rca/programmes/models.py

```diff
@@ -917,6 +926,54 @@ def listing_meta(self):
         # Returns a page 'type' value that's readable for listings,
         return "Programme"
 
+    def get_what_you_will_cover_blocks_accordion_snippet(self):
+        """
+        We cannot directly index the accordion_snippet block in the what_you_will_cover_blocks
+        StreamField. We therefore have to do this manually.
+
+        Ref: https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-callables-and-other-attributes
+        """
+
+        if not self.what_you_will_cover_blocks:
+            return None
+
+        data = self.what_you_will_cover_blocks.get_prep_value()
+        accordion_snippet_values = [
+            item["value"] for item in data if item["type"] == "accordion_snippet"
+        ]
+
+        if not accordion_snippet_values:
+            return None
+
+        accordion_snippet_content = AccordionSnippet.objects.filter(
+            pk__in=accordion_snippet_values
+        ).values_list("heading", "preview_text", "body")
+        return "\n".join(list(chain(*accordion_snippet_content)))
+
+    def get_requirements_blocks_accordion_snippet(self):
+        """
+        We cannot directly index the accordion_snippet block in the requirements_blocks
+        StreamField. We therefore have to do this manually.
+
+        Ref: https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-callables-and-other-attributes
+        """
+
+        if not self.requirements_blocks:
+            return None
+
+        data = self.requirements_blocks.get_prep_value()
+        accordion_snippet_values = [
+            item["value"] for item in data if item["type"] == "accordion_snippet"
+        ]
+
+        if not accordion_snippet_values:
+            return None
+
+        accordion_snippet_content = AccordionSnippet.objects.filter(
+            pk__in=accordion_snippet_values
+        ).values_list("heading", "preview_text", "body")
+        return "\n".join(list(chain(*accordion_snippet_content)))
+
```

Comment:
**Suggestion**

Potentially worth refactoring this to be a shared function or at least method. E.G

```python
def get_accordion_snippet_content(self, stream_field):
    """
    Retrieve content from accordion_snippet blocks in the given StreamField.
    """
    if not stream_field:
        return None

    data = stream_field.get_prep_value()
    ... etc etc

    return "\n".join(chain(*accordion_snippet_content))

def get_what_you_will_cover_blocks_accordion_snippet(self):
    """
    Retrieve content from accordion_snippet blocks in the what_you_will_cover_blocks StreamField.
    """
    return self.get_accordion_snippet_content(self.what_you_will_cover_blocks)

def get_requirements_blocks_accordion_snippet(self):
    """
    Retrieve content from accordion_snippet blocks in the requirements_blocks StreamField.
    """
    return self.get_accordion_snippet_content(self.requirements_blocks)

Reply:
I agree, that's much better! I've refactored the code in b4d80600